src/composables/fungibleTokens.ts

```diff
@@ -0,0 +1,310 @@
+          });
+
+        tokenBalances.value[address] = balances;
+      } catch (error) {
+        handleUnknownError(error);
+      }
+    }));
+  }
+
+  async function createOrChangeAllowance(contractId: string, amount: number | string) {
+    const aeSdk = await getAeSdk();
+    const account = getLastActiveProtocolAccount(PROTOCOL_AETERNITY);
+    const selectedToken = tokenBalances.value?.[account?.address!]
+      ?.find((token) => token?.contractId === contractId);
+
+    const tokenContract = await aeSdk.initializeContract({
+      aci: FungibleTokenFullInterfaceACI,
+      address: selectedToken?.contractId as any,
+    });
+
+    const { decodedResult } = await tokenContract.allowance({
+      from_account: account?.address,
+      for_account: tippingContractAddresses?.value?.tippingV2?.replace('ct_', 'ak_'),
+    });
+
+    const allowanceAmount = (decodedResult !== undefined)
+      ? new BigNumber(decodedResult)
+        .multipliedBy(-1)
+        .plus(toShiftedBigNumber(amount, selectedToken?.decimals!))
+        .toNumber()
+      : toShiftedBigNumber(amount, selectedToken?.decimals!)
+        .toNumber();
+
+    const getContractFunction = (tokenContract.methods as any)[
+      decodedResult !== undefined ? 'change_allowance' : 'create_allowance'
+    ];
+
+    return getContractFunction(
+      tippingContractAddresses.value?.tippingV2?.replace(
+        `${Encoding.ContractAddress}_`,
+        `${Encoding.AccountAddress}_`,
+      ),
+      allowanceAmount,
+    );
+  }
+
+  async function getContractTokenPairs(
+    address: Encoded.ContractAddress,
+  ): Promise<Record<string, IToken>> {
+    try {
+      const aeSdk = await getAeSdk();
+      const account = getLastActiveProtocolAccount(PROTOCOL_AETERNITY);
+      const tokenContract = await aeSdk.initializeContract({
+        aci: AedexV2PairACI,
+        address,
+      });
+
+      const [
+        { decodedResult: balances },
+        { decodedResult: balance },
+        { decodedResult: token0 },
+        { decodedResult: token1 },
+        { decodedResult: reserves },
+        { decodedResult: totalSupply },
+      ] = await Promise.all([
+        tokenContract.balances(),
+        tokenContract.balance(account?.address),
+        tokenContract.token0(),
+        tokenContract.token1(),
+        tokenContract.get_reserves(),
+        tokenContract.total_supply(),
+      ]);
+
+      return {
+        token0: {
+          ...availableTokens.value?.[token0],
+          amount: calculateSupplyAmount(
+            balance,
+            totalSupply,
+            reserves.reserve0,
+          ),
+        },
+        token1: {
+          ...availableTokens.value?.[token1],
+          amount: calculateSupplyAmount(
+            balance,
+            totalSupply,
+            reserves.reserve1,
+          ),
+        },
+        totalSupply,
+        balance,
+        balances,
+      };
+    } catch (error) {
+      return {};
+    }
+  }
+
+  async function transfer(
```

Comment:
Previously this method was used as `dispatch('fungibleTokens/transfer', ...)` so it was clear what kind of transfer we are making. Now the method `transfer` is not clear enough in my opinion. I suggest to rename it to `transferToken`.